informer/src/informer_service/informer_service_handler.rs

```diff
@@ -92,8 +110,24 @@ impl InformerService {
         });
     }
 
-    pub fn add_new_connection(&mut self, connection: Arc<Connection>) {
-        self.connections.push(connection);
+    pub fn add_new_subscription(&mut self, subscription: Arc<Subscription>) {
+        self.subscriptions.push(subscription);
+    }
+
+    pub fn remove_subscription(&mut self, sub_id: SubscriptionId) {
+        let index = self
+            .subscriptions
+            .iter()
+            .position(|subscription| subscription.subscription_id == sub_id)
+            .expect("The index optionality is already checked while receiving the subscription ID ");
+        let removed_subscription = self.subscriptions.remove(index);
+        if let SubscriptionState::Subscribed = removed_subscription.cold_state {
+            removed_subscription.state.store(false, SeqCst);
+        }
```

Comment:
Would you explain why you add the if statme to me?
How about setting the state to false always?

```suggestion
            removed_subscription.state.store(false, SeqCst);
```